pkg/network/admitter/netiface.go

```diff
@@ -97,3 +100,101 @@ func validateInterfaceNameUnique(field *k8sfield.Path, spec *v1.VirtualMachineIn
 	}
 	return causes
 }
+
+func validateInterfacesFields(field *k8sfield.Path, spec *v1.VirtualMachineInstanceSpec) []metav1.StatusCause {
+	var causes []metav1.StatusCause
+	for idx, iface := range spec.Domain.Devices.Interfaces {
+		causes = append(causes, validateInterfaceNameFormat(field, idx, iface)...)
+		causes = append(causes, validateInterfaceModel(field, idx, iface)...)
+		causes = append(causes, validateMacAddress(field, idx, iface)...)
+		causes = append(causes, validatePciAddress(field, idx, iface)...)
+	}
+	return causes
+}
+
+func validateInterfaceNameFormat(field *k8sfield.Path, idx int, iface v1.Interface) []metav1.StatusCause {
+	isValid := regexp.MustCompile(`^[A-Za-z0-9-_]+$`).MatchString
+	if !isValid(iface.Name) {
+		return []metav1.StatusCause{{
+			Type:    metav1.CauseTypeFieldValueInvalid,
+			Message: "Network interface name can only contain alphabetical characters, numbers, dashes (-) or underscores (_)",
+			Field:   field.Child("domain", "devices", "interfaces").Index(idx).Child("name").String(),
+		}}
+	}
+	return nil
+}
+
+var validInterfaceModels = map[string]struct{}{
+	"e1000":    {},
+	"e1000e":   {},
+	"ne2k_pci": {},
+	"pcnet":    {},
+	"rtl8139":  {},
+	v1.VirtIO:  {},
+}
+
+func validateInterfaceModel(field *k8sfield.Path, idx int, iface v1.Interface) []metav1.StatusCause {
+	if iface.Model != "" {
+		if _, exists := validInterfaceModels[iface.Model]; !exists {
+			return []metav1.StatusCause{{
+				Type: metav1.CauseTypeFieldValueNotSupported,
+				Message: fmt.Sprintf(
+					"interface %s uses model %s that is not supported.",
+					field.Child("domain", "devices", "interfaces").Index(idx).Child("name").String(),
+					iface.Model,
+				),
+				Field: field.Child("domain", "devices", "interfaces").Index(idx).Child("model").String(),
+			}}
+		}
+	}
+	return nil
+}
+
+func validateMacAddress(field *k8sfield.Path, idx int, iface v1.Interface) []metav1.StatusCause {
+	var causes []metav1.StatusCause
+	if iface.MacAddress != "" {
+		mac, err := net.ParseMAC(iface.MacAddress)
+		if err != nil {
+			causes = append(causes, metav1.StatusCause{
+				Type: metav1.CauseTypeFieldValueInvalid,
+				Message: fmt.Sprintf(
+					"interface %s has malformed MAC address (%s).",
+					field.Child("domain", "devices", "interfaces").Index(idx).Child("name").String(),
+					iface.MacAddress,
+				),
+				Field: field.Child("domain", "devices", "interfaces").Index(idx).Child("macAddress").String(),
```

Comment:
nit: I think the field path string can be reused

Reply:
Same answer as before.